lib/LLVMIRCodeGen/BundleSaver.cpp

```diff
@@ -84,6 +138,160 @@ void BundleSaver::saveWeights(llvm::StringRef weightsFileName) {
   weightsFile.close();
 }
 
+void BundleSaver::saveHeader(llvm::StringRef headerFileName) {
+  std::error_code EC;
+  llvm::raw_fd_ostream headerFile(headerFileName, EC,
+                                  llvm::sys::fs::OpenFlags::F_Text);
+  CHECK(!EC) << "Could not open header file!";
+  auto bundleName = irgen_->getBundleName();
+  auto bundleNameUpper = llvm::StringRef(bundleName).upper();
+  auto headerDefine = "_GLOW_BUNDLE_" + bundleNameUpper + "_H";
+  auto commonDefine = "_GLOW_BUNDLE_COMMON_DEFS";
+  auto constMemSize = irgen_->getAllocationsInfo().constantWeightVarsMemSize_;
+  auto mutableMemSize = irgen_->getAllocationsInfo().mutableWeightVarsMemSize_;
+  auto activationsMemSize = irgen_->getAllocationsInfo().activationsMemSize_;
+  auto memAlignSize = TensorAlignment;
+  auto totMemSize = constMemSize + mutableMemSize + activationsMemSize;
```

Comment:
"totMemSize" is not used (or intended) for allocation memory. It is merely printed as a comment in the header file (I found it very useful since I can verify quickly if a model fits or not on an MCU).

For dynamic API the memory sizes are defined separately in the bundleConfig.
For static API the memory sizes are printed separately as macros (e.g. for lenet_mnist):
#define LENET_MNIST_CONSTANT_MEM_SIZE     1724672
#define LENET_MNIST_MUTABLE_MEM_SIZE      3200
#define LENET_MNIST_ACTIVATIONS_MEM_SIZE  57600